protocol/x/prices/keeper/market.go

```diff
@@ -57,6 +57,13 @@ func (k Keeper) CreateMarket(
 		)
 	}
 	currencyPairStr := currencyPair.String()
+	_, err = k.MarketMapKeeper.GetMarket(ctx, currencyPairStr)
```

Comment:
Are there tests for this new logic?

Reply:
added an explicit check to market_test.go that error is not nil when creating new market before it exists in market map